### PR TITLE
🛀 Upgrade everything!

### PR DESCRIPTION
**Pull request by swashcap**
* **Problem:** _coins-browser-test_ is out of date
* **Solution:**
  * Remove unused requirements
  * Bump existing requirements' versions
  * Add ESLint, style guide rules. Make everything pass linting
  * Put linting in pre-commit hook
* **Testing:** Ensure all tests (`test/*.js`) pass

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -5,7 +5,8 @@
   "main": "index.js",
   "scripts": {
     "clean": "rm selenium-server-standalone-3.0.1.jar",
-    "install": "curl -O https://selenium-release.storage.googleapis.com/3.0/selenium-server-standalone-3.0.1.jar",
+    "install": "stat selenium-server-standalone-3.0.1.jar || curl -O https://selenium-release.storage.googleapis.com/3.0/selenium-server-standalone-3.0.1.jar",
```
Comment (cdaringe) [inline package.json]: 😍

File: test/lib/client.js [hunk 2]
```diff
@@ -1,34 +1,33 @@
...
-        if(!err) {
-            resolve();
-        } else {
-            reject(err);
-        }
-    });
-}).catch(function(err) {
-    console.error(err);
-    throw err;
+const clientReady = new Promise((resolve, reject) => {
```
Comment (cdaringe) [inline test/lib/client.js]: #pify, sometime

**Comment by cdaringe** [comment]
@swashcap, looks great!  looks like it took forever! :shipit:

File: test/lib/client.js (hunk 2 above)
Comment (swashcap) [inline test/lib/client.js]: 4reals. Haven't updated Webdriver.io _yet_, which might make this totally unnecessary.

**Comment by swashcap** [comment]
@cdaringe Yeah…😞 `--fix` can’t get everything.

**Comment by drewlandis** [comment]
ditto, this looks really nice!

**Pull request merged**